bindings/java/src/org/sleuthkit/datamodel/OsAccountManager.java

```diff
@@ -55,121 +54,123 @@
 	}
 
 	/**
-	 * Creates an OS account with given unique id or given realm and login name.
-	 * If an account already exists with the given id or realm/login, then the
+	 * Creates an OS account with given unique id and given realm id.
+	 * If an account already exists with the given id, then the
 	 * existing OS account is returned.
 	 *
 	 * @param uniqueAccountId Account sid/uid.
-	 * @param loginName       Login name.
-	 * @param realmName       Realm within which the accountId or login name is
-	 *                        unique.
-	 * @param host            Host for the realm, may be null.
+	 * @param realmId       Realm id.
+	 * 
+	 * @return OsAccount.
+	 *
+	 * @throws TskCoreException If there is an error in creating the OSAccount.
+	 *
+	 */
+	OsAccount createOsAccount(String uniqueAccountId,  long realmId) throws TskCoreException {
+
+		// ensure unique id is provided
+		if (Strings.isNullOrEmpty(uniqueAccountId)) {
+			throw new IllegalArgumentException("Cannot create OS account with null uniqueId.");
+		}
+		
+		try (CaseDbConnection connection = this.db.getConnection();) {
+
+			OsAccountRealm realm = db.getOsAccountRealmManager().getRealm(realmId, connection);
```

Comment:
@raman-bt and @bcarrier, this may be a fine point, but do we want to create coupling between the OsAccountManager and the OsRealmManager? It seems to be that it would be a cleaner design, at least according to my personal understanding of classical design principles, to have a clear separation of concerns / no coupling.  The client could pass in an OsAccountRealm obtained by making the call themselves. Such an approach would also make error handling easier when all exceptions are TskCoreExceptions. With this code as is, the user does not know whether an exception is coming from the code to look up the realm, or the code to create the account. The client's response may need to be different for each error condition, but that is not possible here.

Reply:
@raman-bt and @bcarrier, actually, I see that this method is package scope but has no callers. If we think we are going to need it later, perhaps we should comment it out to show our intent? In any case. I think my question in the previous comment still applies, so that I can orient myself better to what our goals are.

Reply:
w.r.t to coupling, A realm is a part of an Account so there is going to be some "coupling" in the form of using apis.

In this instance however I agree that instead of the realmRowID, we can pass the realm itself.  Purely from efficiency point of view, we are wasting cycles - getting the realm, passing the id, then getting the realm again downstream. 

There are other instances where the doc specifies realmRowId as an argument but I replaced it with the  accountRealm.  I shall do the same here.

It is indeed not used from anywhere yet.  I don't know what usecase Brian has in mind.

Reply:
"Purely from efficiency point of view, we are wasting cycles - getting the realm, passing the id, then getting the realm again downstream."

I'm sorry, I don't follow this argument. I assumed we would pass in an OsAccountRealm object instead of an ID. What am I missing?

Reply:
I did change it to pass the OsAccountRealm instead of the id.   I'm in  agreement with you albeit for a different reason.  

I meant passing in the id when we already have the object in hand, and then the API fetching the object again is inefficient.